### ISSUE TITLE
Alpha AppImage doesn't work on Ubuntu 14.04

### ISSUE THREAD

**Issue by danpla** [mypaint/mypaint#966]
The AppImage ([v2.0.0-alpha.5](https://github.com/mypaint/mypaint/releases/tag/v2.0.0-alpha.5)) doesn't work on Ubuntu 14.04.

```
DIR: /tmp/.mount_MyPainXuRZ9K
AILIBDIR=/tmp/tmp.hdpgKAKefj
System stdc++ library: "/usr/lib/x86_64-linux-gnu/libstdc++.so.6"
System stdc++ library version: "3.4.25"
Bundled stdc++ library version: "3.4.19"
Newest stdc++ library version: "3.4.25"
Using system stdc++ library
atomiclib: /usr/lib/x86_64-linux-gnu/libatomic.so.1
Checking versions of library "libfontconfig"
  system library: "/usr/lib/x86_64-linux-gnu/libfontconfig.so.1.8.0" (libfontconfig.so.1.8.0)
  system library version: 1.8.0
/tmp/.mount_MyPainXuRZ9K/apprun-helper.sh: line 105: realpath: command not found
  ailib: ""
  bundled library version: 
  newest library version: "1.8.0"
Removing bundled "libfontconfig"
removed '/tmp/tmp.hdpgKAKefj/libfontconfig.so.1'
removed '/tmp/tmp.hdpgKAKefj/libfontconfig.so.1.11.1'
Checking versions of library "libfreetype"
  system library: "/usr/lib/x86_64-linux-gnu/libfreetype.so.6.11.1" (libfreetype.so.6.11.1)
  system library version: 6.11.1
/tmp/.mount_MyPainXuRZ9K/apprun-helper.sh: line 105: realpath: command not found
  ailib: ""
  bundled library version: 
  newest library version: "6.11.1"
Removing bundled "libfreetype"
removed '/tmp/tmp.hdpgKAKefj/libfreetype.so.6'
removed '/tmp/tmp.hdpgKAKefj/libfreetype.so.6.14.0'
GTK_PATH=/tmp/.mount_MyPainXuRZ9K/usr/lib/gtk-2.0
GTK_IM_MODULE_FILE=/tmp/.mount_MyPainXuRZ9K/usr/lib/gtk-2.0:/tmp/.mount_MyPainXuRZ9K/usr/lib/gtk-2.0
PANGO_LIBDIR=/tmp/.mount_MyPainXuRZ9K/usr/lib
/tmp/.mount_MyPainXuRZ9K/apprun-helper.sh: line 131: /tmp/.mount_MyPainXuRZ9K/startup_scripts/*.sh: No such file or directory
INFO: mypaint: Installation layout: conventional POSIX-like structure with prefix u'/tmp/.mount_MyPainXuRZ9K/usr'
Traceback (most recent call last):
  File "/tmp/.mount_MyPainXuRZ9K/usr/bin/mypaint", line 464, in <module>
    = get_paths()
  File "/tmp/.mount_MyPainXuRZ9K/usr/bin/mypaint", line 230, in get_paths
    from lib import fileutils
  File "/tmp/.mount_MyPainXuRZ9K/usr/lib/mypaint/lib/fileutils.py", line 26, in <module>
    import lib.helpers
  File "/tmp/.mount_MyPainXuRZ9K/usr/lib/mypaint/lib/helpers.py", line 26, in <module>
    from . import mypaintlib
  File "/tmp/.mount_MyPainXuRZ9K/usr/lib/mypaint/lib/mypaintlib.py", line 26, in <module>
    _mypaintlib = swig_import_helper()
  File "/tmp/.mount_MyPainXuRZ9K/usr/lib/mypaint/lib/mypaintlib.py", line 22, in swig_import_helper
    _mod = imp.load_module('_mypaintlib', fp, pathname, description)
ImportError: /tmp/tmp.hdpgKAKefj/libpangoft2-1.0.so.0: undefined symbol: FcWeightFromOpenTypeDouble

```

**Comment by briend** [comment]
Can you see if this helps?
https://bbs.archlinux.org/viewtopic.php?id=235716

They had success by uninstalling fontconfig-ubuntu and installing fontconfig

**Comment by danpla** [comment]
The fontconfig is already installed (I believe it's one of the pre-installed packages on every Ubuntu system).

Based on the backtrace, I assumed that the shipped `libpangoft` expects a newer version of `libfontconfig` than the one installed on the system. `FcWeightFromOpenTypeDouble` is only available in `libfontconfig` versions shipped with 16.04 or newer (see `fontconfig.h` file in `libfontconfig1-dev` package). However, later I found that the right version of `libfontconfig` is already in AppImage.

The actual problem is:

```
realpath: command not found
```
Because of that, the `apprun-helper.sh` script cannot detect the shipped version of `libfontconfig` (and other libraries as well). The fix is to replace `realpath` in the script with `readlink -f`.

Why this happens? Because `realpath` should be installed separately on Ubuntu 14.04 and older. On 16.06 and newer, `relpath` is a part of `coreutils` package. For more info, see [What's the difference between “realpath” and “readlink -f”](https://unix.stackexchange.com/questions/136494/whats-the-difference-between-realpath-and-readlink-f).

**Comment by jplloyd** [comment]
This issue is now fixed in the [latest appimage builds](https://github.com/mypaint/mypaint-appimage/releases/tag/continuous).
